### PR TITLE
updated combatant list so can edit current hp and do math with it

### PR DESCRIPTION
**Pull request by TKDMaster12**
@Mandalorian007 
updated combatant list so can edit current hp and do math with it
made functions dynamic so they will work with the other edits when added

want to figure out how to get combatants props from encouterDrawers and fill state with that instead of fetching data again. Any ideas?

can't figure out why data wont save to database. It updates information, you can refresh the page go to other pages it still shows new values. But if you stop program and start again it goes back to old data.

not sure why package.json rearranged everything
should have only added "babel-preset-stage-0": "^6.24.1" at the end

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +32,7 @@
     "babel-loader": "^6.2.7",
     "babel-polyfill": "^6.16.0",
     "babel-preset-es2015": "^6.18.0",
-    "babel-preset-react": "^6.16.0"
+    "babel-preset-react": "^6.16.0",
+    "babel-preset-stage-0": "^6.24.1"
```
Comment (Mandalorian007) [inline package.json]: shouldn't this be version 6.16.0 like the rest of the babel imports?

File: src/main/js/combatantList.js [hunk 2]
```diff
@@ -46,6 +55,52 @@ class CombatantListUnstyled extends React.Component {
     this.setState({open: false})
   }
 
+  componentDidMount() {
+    this.refreshCombatantsState();
+  }
+
+  refreshCombatantsState() {
```
Comment (Mandalorian007) [inline src/main/js/combatantList.js]: This shouldn't be duplicated from the encounter drawer

File: src/main/js/combatantList.js [hunk 3]
```diff
@@ -46,6 +55,52 @@ class CombatantListUnstyled extends React.Component {
...
+    fetch(`http://localhost:8080/combatants`)
+      .then(results => results.json())
+      .then(data => this.setState({combatants: data}));
+  }
+
+  updateCombatant(index, data) {
```
Comment (Mandalorian007) [inline src/main/js/combatantList.js]: This function needs to be pulled up the EncounterDrawer and passed down

File: src/main/js/combatantList.js [hunk 4]
```diff
@@ -63,7 +118,7 @@ class CombatantListUnstyled extends React.Component {
               </TableRow>
             </TableHead>
             <TableBody>
-              {this.props.combatants.map(combatant => {
```
Comment (Mandalorian007) [inline src/main/js/combatantList.js]: this should stay props and let the top level manage combatant state

File: package.json (hunk 1 above)
Comment (Mandalorian007) [reply package.json]: Also don't care if the order is re-shuffled must be something you run locally, but we can roll with this

**Comment by TKDMaster12** [comment]
@Mandalorian007 
okay. I moved one of the functions over to encounterDrawer
Deleted the copied files and used props to reference them instead

**Comment by TKDMaster12** [comment]
@Mandalorian007 
I removed the index and used combantId . I build all the data for teh updatecompatant function inside the handlecahnge function. 

For some reason it is still not saving the data. if you stop spring-boot and then start up again you see the old data

File: src/main/js/encounterDrawer.js [hunk 5]
```diff
@@ -87,9 +88,36 @@ class EncounterDrawerUnstyled extends React.Component {
...
+        return item;
+    });
+
+    this.setState({
+       combatants: newState
+    });
+
+    if (!isNaN(value))
+    {
+        let url = 'http://localhost:8080/combatants/' + combatantId;
```
Comment (Mandalorian007) [inline src/main/js/encounterDrawer.js]: Let's leave the url building in the updateCombatant so all the modification methods hide the url.

File: src/main/js/encounterDrawer.js [hunk 6]
```diff
@@ -87,9 +88,36 @@ class EncounterDrawerUnstyled extends React.Component {
...
+          method: 'PATCH',
+          body: JSON.stringify(data),
+          headers: {
+            'Content-Type': 'application/json'
+          }
+      }).catch(err => err)
+      .then(() => this.refreshCombatantsState());
+  }
+
+  handleChange(combatantId, dataType, value) {
```
Comment (Mandalorian007) [inline src/main/js/encounterDrawer.js]: can you move the handle change to the combatantList component and then pass the updateCombatant in?

File: src/main/js/encounterDrawer.js [hunk 7]
```diff
@@ -87,9 +88,36 @@ class EncounterDrawerUnstyled extends React.Component {
...
+    let data;
+    const newState = this.state.combatants.map(item => {
```
Comment (Mandalorian007) [inline src/main/js/encounterDrawer.js]: I'm not sure I understand this logic are you calling the url with a patch like:

{
   currentHp: 32
}

File: src/main/js/encounterDrawer.js (hunk 6 above)
Comment (Mandalorian007) [reply src/main/js/encounterDrawer.js]: Trying to isolate the combatantUpdate operations and the individual component flow work.  Considering adding redux to simplify state management longer term

File: src/main/js/encounterDrawer.js (hunk 7 above)
Comment (TKDMaster12) [reply src/main/js/encounterDrawer.js]: yes that is exactly whats happening. 
I am using your CompatnatApI
@PatchMapping("combatants/{combatantId}")
    public Combatant updateCombatant

File: src/main/js/encounterDrawer.js (hunk 6 above)
Comment (TKDMaster12) [reply src/main/js/encounterDrawer.js]: If move the handleChange to the combatantList file, then how do I setState
    this.setState({
       combatants: newState
    });

is there a built in function that i can update parent state from child?

File: src/main/js/encounterDrawer.js (hunk 7 above)
Comment (Mandalorian007) [reply src/main/js/encounterDrawer.js]: Okay just built and tested your branch on my computer.

Results:

I hit the API directly to test a modification to the currentHp: Used PostMan
localhost:8080/combatants/326
{
	"currentHp": 1
}

Successfully updated the hp to 1


I launched the UI and played with the text field.  Changing the text field to a new number and refreshing the page (via web browser)  kept the change.

Using math in the field didn't seem to have any effect and it just keep the hit points at whatever the number was.  example: +50 became 50

**Comment by TKDMaster12** [comment]
@Mandalorian007 
add function for delete
Added perception field
Added style to the table

delete button works.  Math not working with out setting state. Seeing some kind of lag in changehandle with out setting the state.

**Comment by TKDMaster12** [comment]
@Mandalorian007 
Finished styling 
added color effect for current hp
I notice lag when updating any fields let me know if that is my computer or the program.

If everything looks good to you, I am ready to merge

**Pull request merged**